### PR TITLE
Metadata (select) support for the Supabase data provider

### LINKED ISSUE #1489: [FEAT] Metadata support for the Supabase data provider
Allow overriding `select('*')` via metadata in the Supabase data provider.

Example of how you can use it:
```
const { tableProps } = useTable<IEvent>({
  permanentFilter: [{ field: "devices.tenant_id", operator: "eq", value: tenant }],
  metaData: {
    select: '*, devices!inner(tenant_id)'
  },
});
```

### PR DESCRIPTION
**Pull request by fspijkerman**
This closes #1489

Metadata (select) support for the Supabase data provider

### CODE REVIEW

**Review by omeraplak — APPROVED** [review]

**Review by salihozdemir — APPROVED** [review]

File: packages/supabase/src/index.ts [hunk 1]
```diff
@@ -62,13 +62,15 @@ const handleError = (error: PostgrestError) => {
...
+        getList: async ({ resource, pagination, filters, sort, metaData }) => {
             const current = pagination?.current || 1;
             const pageSize = pagination?.pageSize || 10;
 
             const query = supabaseClient
                 .from(resource)
-                .select("*", { count: "exact" })
+                .select(metaData?.select ? metaData?.select : "*", {
+                    count: "exact",
+                })
```
Comment (omeraplak) [inline packages/supabase/src/index.ts:73]:
```suggestion
                .select(metaData?.select ?? "*", {
                    count: "exact",
                })
```
How would you like it to be this way?

File: packages/supabase/src/index.ts [hunk 2]
```diff
@@ -91,10 +93,10 @@ const dataProvider = (supabaseClient: SupabaseClient): DataProvider => {
             };
         },
 
-        getMany: async ({ resource, ids }) => {
+        getMany: async ({ resource, ids, metaData }) => {
             const { data, error } = await supabaseClient
                 .from(resource)
-                .select("*")
+                .select(metaData?.select ? metaData?.select : "*")
```
Comment (omeraplak) [inline packages/supabase/src/index.ts:99]:
```suggestion
                .select(metaData?.select ?? "*")
```

File: packages/supabase/src/index.ts [hunk 3]
```diff
@@ -170,10 +172,10 @@ const dataProvider = (supabaseClient: SupabaseClient): DataProvider => {
             };
         },
 
-        getOne: async ({ resource, id }) => {
+        getOne: async ({ resource, id, metaData }) => {
             const { data, error } = await supabaseClient
                 .from(resource)
-                .select("*")
+                .select(metaData?.select ? metaData?.select : "*")
```
Comment (omeraplak) [inline packages/supabase/src/index.ts:178]:
```suggestion
                .select(metaData?.select ?? "*")
```

File: packages/supabase/src/index.ts (hunk 1 above)
Comment (fspijkerman) [reply packages/supabase/src/index.ts:73]: Much better, thanks!

**Review by omeraplak — APPROVED** [review]

**Pull request merged**